components/bms_worker/src/BatteryMonitoring.c

```diff
@@ -417,3 +420,25 @@ void BMS_DischargeLimit(uint16_t relativeSoC) {
         BMS.dischargeLimit = 2.25f*relativeSoC; //linear function for the last 20% of discharge
     }
 }
+
+void BMS_HeatCurrentDischargeLimit(int16_t cellTemp) {
+    if (cellTemp >= 48) {
+        BMS.heatChargeCurrentLimit    = -3.75f * cellTemp + 225;
+    } else if (cellTemp > 60) {
+        BMS.heatDischargeCurrentLimit = 0;
+    } else {
+        BMS.heatDischargeCurrentLimit = MAX_CONTINOUS_DISCHARGE_CURRENT;
+    }
+}
+
+void BMS_HeatCurrentChargeLimit(int16_t cellTemp) {
```

Comment:
lowerCamelCase gain. IE BMS_heatCurrentChargeLimit

Reply:
If you pass the cellTemp as an arg, you should return the calculated value. if not an argm, then you should take and put to the struct from the function